packages/scenario/test/secondary-instances.test.ts

```diff
@@ -703,4 +705,86 @@ describe('Secondary instances', () => {
 			});
 		});
 	});
+
+	describe('basic external secondary instance support', () => {
+		const attachmentFileName = 'xml-attachment.xml';
+		const attachmentURL = `jr://file/${attachmentFileName}` as const;
+		const formTitle = 'External secondary instance (XML)';
+		const formDefinition = html(
+			head(
+				title(formTitle),
+				model(
+					mainInstance(t('data id="external-secondary-instance-xml"', t('first'))),
+
+					t(`instance id="external-xml" src="${attachmentURL}"`),
+
+					bind('/data/first').type('string')
+				)
+			),
+			body(
+				// Define a select using value and label references that don't exist in the secondary instance
+				select1Dynamic(
+					'/data/first',
+					"instance('external-xml')/instance-root/instance-item",
+					'item-value',
+					'item-label'
+				)
+			)
+		);
+
+		let resourceService: JRResourceService;
+
+		beforeEach(() => {
+			resourceService = new JRResourceService();
+		});
+
+		afterEach(() => {
+			resourceService.reset();
+		});
+
+		it('supports external secondary instances (XML, file system fixture)', async () => {
+			resourceService.activateFixtures(
+				r(attachmentFileName).toAbsolutePath().getParent().toString(),
+				['file']
+			);
+
+			const scenario = await Scenario.init(formTitle, formDefinition, {
+				resourceService,
+			});
+
+			scenario.answer('/data/first', 'a');
+
+			expect(scenario.answerOf('/data/first')).toEqualAnswer(stringAnswer('a'));
+		});
+
+		// This test is redundant to the one above, but demonstrates how we could
+		// define form attachments inline, like we do form definitions—a pattern
+		// worth considering if we want to expand external secondary instance
```

Comment:
The philosophy in JR has been to test the basic functionality that is specific to external data attachments (especially their parsing) and then to favor tests on internal data instances for functionality that applies to all secondary instances.